lib/elixir/lib/kernel.ex

```diff
@@ -2872,6 +2872,8 @@ defmodule Kernel do
         :lists.foldr(fn x, acc ->
           quote do: :erlang.or(unquote(comp(left, x)), unquote(acc))
         end, comp(left, h), t)
+      {:++, _meta, [h, t]} ->
+        quote do: :erlang.or(unquote(left) in unquote(t), unquote(left) in unquote(h))
```

Comment:
Same reason for the order as below, but I'm not 100% sure I know what change in semantics you mean. Is it that `right` will be evaluated before `left` in this case?